### PR TITLE
fix consul stanza parsing when a configuration directory is used

### PR DESCRIPTION
**Pull request by langmartin**
When parsing a configuration directory, consul, vault, and autopilot stanzas are parsed correctly only if they are in the last file loaded. This change restores correct default merging behavior.

### CODE REVIEW

**Comment by langmartin** [comment]
Fixes #5795 and #5816

File: command/agent/config_parse.go [hunk 1]
```diff
@@ -14,6 +14,15 @@ import (
 	"github.com/hashicorp/nomad/nomad/structs/config"
 )
 
+// ParseConfigDefault returns the configuration base
+func ParseConfigDefault() *Config {
+	return &Config{
```
Comment (preetapan) [inline command/agent/config_parse.go]: Are there more than just these three that need to have defaults set correctly?

Comment (langmartin) [reply command/agent/config_parse.go]: These are the only three where the defaults are required before decoding, other defaults are set in `command.readConfig`

File: command/agent/config.go [hunk 2]
```diff
@@ -1461,16 +1479,22 @@ func LoadConfig(path string) (*Config, error) {
...
 
+	defaults := ParseConfigDefault()
+
 	if fi.IsDir() {
-		return LoadConfigDir(path)
+		config, err := LoadConfigDir(path)
+		if err != nil {
+			return nil, err
+		}
+		return defaults.Merge(config), nil
```
Comment (notnoop) [inline command/agent/config.go]: nitpick: I would consider moving this logic into `LoadConfigDir`.

**Pull request merged**